packages/deployment/lib/lambda.js

```diff
@@ -35,23 +59,28 @@ class UpdatedLambda extends Lambda {
    * @param {Object} lambda - the lambda object
    * @returns {Promise} returns the updated lambda object
    */
-  zipLambda(lambda) {
+  async zipLambda(lambda) {
     let msg = `Zipping ${lambda.local}`;
     // skip if the file with the same hash is zipped
+    // and is a valid zip file
     if (fs.existsSync(lambda.local)) {
-      return Promise.resolve(lambda);
+      try {
+        await this.validateZipFile(lambda);
+        return Promise.resolve(lambda);
+      }
+      catch (e) {
+        console.log(`${lambda.local} appears to be an invalid zip file, and will be re-built`);
```

Comment:
Do we actually retry it if it's invalid?

Reply:
@laurenfrederick yes, the original code was skipping regeneration on L69, if the validator throws an error it logs and continues instead.

Reply:
💯